pkg/controllers/state/statenode.go

```diff
@@ -185,7 +185,7 @@ func (in *StateNode) ValidateDisruptable(ctx context.Context, kubeClient client.
 	if in.Nominated() {
 		return nil, fmt.Errorf("state node is nominated for a pending pod")
 	}
-	if _, ok := in.Annotations()[v1beta1.DoNotDisruptAnnotationKey]; ok {
+	if val, ok := in.Annotations()[v1beta1.DoNotDisruptAnnotationKey]; ok && val == "true" {
```

Comment:
Do you even need to check ok? If you access the map and it's empty, it should be "", so the comparison will still work

Reply:
Agreed. Updated.